app/controllers/application_controller.rb

```diff
@@ -10,6 +10,16 @@ class ApplicationController < ActionController::Base
   before_action :set_locale
 
   rescue_from ActiveRecord::RecordNotFound, with: :render_404
+  before_action :set_csp
+
+  def set_csp
```

Comment:
Why not use the [secure_headers gem](https://github.com/twitter/secureheaders)? This seems pretty hacky.